### ISSUE TITLE
Fake bot plugin

### ISSUE THREAD

**Issue by azurit** [coreruleset/coreruleset#2228]
### Motivation

As we all know, many bots are fakeing `User-Agent` header to look like a, i.e., Googlebot. In lots of cases, users are trusting this header and, for example, are not applying so high restrictions for such requests. Anyway, behavior where anyone is fakeing anything to get something more is considered bad (or should be). This plugin is able to detect and block such requests.

### Proposed solution

This is an alternative and more general version of [Fake Googlebot plugin](https://github.com/coreruleset/coreruleset/issues/2227).

As @theMiddleBlue [suggested](https://github.com/coreruleset/coreruleset/issues/2227#issuecomment-939499768), this version of plugin is using DNS PTR records for checking if a bot is real or fake. Plugin consists of these parts:
 - blocking rule
 - Lua script

Currently, plugin is able to detect these fake bots:
 - Googlebot
 - Bingbot
 - Facebookbot

Prototype of this plugin is ready.

Plugin registry PR: https://github.com/coreruleset/plugin-registry/pull/4.

**Comment by theMiddleBlue** [comment]
wow, that's true... I didn't think about this kind of problem!

```shell
➜  ~ time host -t PTR 192.165.48.59 1.1.1.1
Host 59.48.165.192.in-addr.arpa not found: 2(SERVFAIL)
0,00s user 0,01s system 0% cpu 3,941 total

➜  ~ time host -t PTR 45.141.84.214 1.1.1.1
Host 214.84.141.45.in-addr.arpa not found: 2(SERVFAIL)
0,00s user 0,00s system 0% cpu 8,516 total
```

weird... why this happens?

**Comment by azurit** [comment]
Plugin registry PR: https://github.com/coreruleset/plugin-registry/pull/4

**Comment by azurit** [comment]
@dune73 @lifeforms --^

**Comment by dune73** [comment]
Repo for new plugin created https://github.com/coreruleset/fake-bot-plugin
@azurit has been made maintainer.

I have also added the plugin into the registry, which is formally too early, but that way @azurit can get on with the work.
